lib/pi_piper/pin.rb

```diff
@@ -105,8 +100,8 @@ def update_value(new_value)
     # @param [Symbol] state Indicates if and how pull mode must be set when 
     # pin direction is set to :in. Either :up, :down or :offing. Defaults to :off.
     def pull!(state)
-      return nil if @direction != :in
-      fail PiPiper::PinError, "Pin #{@pin} already released" if released?
+     raise PiPiper::PinError, "Unable to use pull-ups : pin direction must be ':in' for this" if 
+        @direction != :in and [:up, :down].include?(state)
```

Comment:
Do we really need the check on state if `@direction == :out`? I thought you can't `pull!` on `:out` pins? Even if the pull is set to `:off`, I think raising an error is fine instead of returning `nil` since this method shouldn't even be called in those cases.